### PR TITLE
Fix #201: `*> ZIO.succeed` -> `.as`

### PR DESCRIPTION
**Pull request by YBogomolov**
Implement https://github.com/zio/zio-intellij/issues/201

### CODE REVIEW

File: src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.jetbrains.plugins.scala.codeInspection.collections._
+import org.jetbrains.plugins.scala.lang.psi.api.expr.ScExpression
+import zio.intellij.inspections._
+import zio.intellij.inspections.zioMethods._
+import zio.intellij.utils.StringUtils._
+
+class SimplifySucceedInspection extends ZInspection(SucceedSimplificationType)
+
+object SucceedSimplificationType extends SimplificationType {
```
Comment (alesavin) [inline src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala]: Why not to make it part of SimplifyZipInspection? For me it is mostly simplification for zip, not succeed

File: src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  override def hint: String = "Replace with .as"
+
+  override def getSimplification(expr: ScExpression): Option[Simplification] = {
+    def simplify(scExpr: ScExpression, arg: ScExpression): SimplificationBuilder =
+      replace(expr).withText(invocationText(scExpr, s"as${arg.getWrappedText}")).highlightAll
+
+    expr match {
+      case qual `.*>` `ZIO.succeed`(_, arg) => Some(simplify(qual, arg))
```
Comment (alesavin) [inline src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala]: Like Nikita mention it is worth to add case for zipLeft as well here

File: src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala (hunk 1 above)
Comment (YBogomolov) [reply src/main/scala/zio/intellij/inspections/simplifications/SimplifySucceedInspection.scala]: Makes sense — I moved code there.

**Comment by alesavin** [comment]
LGTL, but seems we have to fix tests

**Review by myazinn — APPROVED** [review]
Nice work!

File: src/test/scala/zio/inspections/SimplifyZipRightInspectionTest.scala [hunk 3]
```diff
@@ -117,3 +117,29 @@ class SimplifyFlatmapWithZipRightOperatorTest extends ZipRightInspectionTest("*>
...
+    testQuickFixes(text, result, hint)
+  }
+
+  def test_zipRight_operator_to_succeed(): Unit = {
+    z(s"""${START}f("Fernando Fader") *> ZIO.succeed("Carlos Alonso")$END""").assertHighlighted()
+    val text   = z("""f("Fernando Fader") *> ZIO.succeed("Carlos Alonso")""")
+    val result = z("""f("Fernando Fader").as("Carlos Alonso")""")
+    testQuickFixes(text, result, hint)
+  }
+
```
Comment (alesavin) [inline src/test/scala/zio/inspections/SimplifyZipRightInspectionTest.scala:144]: nit: we really need this newline?

**Review by alesavin — APPROVED** [review]

File: src/test/scala/zio/inspections/SimplifyZipRightInspectionTest.scala (hunk 3 above)
Comment (YBogomolov) [reply src/test/scala/zio/inspections/SimplifyZipRightInspectionTest.scala:144]: My thoughts — nope, but that's the style other tests are written.

**Pull request merged**